src/test/java/uk/gov/register/resources/VerifiableLogResourceTest.java

```diff
@@ -1,138 +1,73 @@
+        when(vlServiceMock.getRegisterProof()).thenReturn(expectedProof);
 
-        RegisterProof registerProof = verifiableLogResource.registerProof();
+        VerifiableLogResource vlResource = new VerifiableLogResource(vlServiceMock);
+        RegisterProof actualProof = vlResource.registerProof();
 
-        assertThat(registerProof.getProofIdentifier(), equalTo("merkle:sha-256"));
-        assertThat(registerProof.getRootHash(), equalTo("d3d33f57b033d18ad11e14b28ef6f33487410c98548d1759c772370dfeb6db11"));
+        verify(vlServiceMock, times(1)).getRegisterProof();
+        assertThat(actualProof.getProofIdentifier(), equalTo(expectedProof.getProofIdentifier()));
+        assertThat(actualProof.getRootHash(), equalTo(expectedProof.getRootHash()));
     }
 
     @Test
-    public void shouldReturnValidRegisterProofForTwoEntries() throws Exception {
-        EntryQueryDAO entryDAO = entryDAOForEntries(entry1, entry2);
-        VerifiableLogResource verifiableLogResource = new VerifiableLogResource(entryDAO, new DoNothing());
-
-        RegisterProof registerProof = verifiableLogResource.registerProof();
-
-        assertThat(registerProof.getProofIdentifier(), equalTo("merkle:sha-256"));
-        assertThat(registerProof.getRootHash(), equalTo("e869291e3017a7b1dd6b16af0b556d75378bef59486f1a7f53586b3ca86aed09"));
-    }
-
-    @Test
-    public void shouldReturnValidRegisterProofForThreeEntries() throws Exception {
-        EntryQueryDAO entryDAO = entryDAOForEntries(entry1, entry2, entry3);
-        VerifiableLogResource verifiableLogResource = new VerifiableLogResource(entryDAO, new DoNothing());
-
-        RegisterProof registerProof = verifiableLogResource.registerProof();
-
-        assertThat(registerProof.getProofIdentifier(), equalTo("merkle:sha-256"));
-        assertThat(registerProof.getRootHash(), equalTo("6b85b168f7c5f0587fc22ff4ba6937e61b33f6e89b70eed53d78d895d35dc9c3"));
+    public void shouldUseServiceToGetEntryProof() throws NoSuchAlgorithmException {
+        int entryNumber = 2;
+        int totalEntries = 5;
+        List<String> expectedAuditPath = Arrays.asList(sampleHash1, sampleHash2);
+
+        EntryProof expectedProof = new EntryProof("3", expectedAuditPath);
+        VerifiableLogService vlServiceMock = mock(VerifiableLogService.class);
+        when(vlServiceMock.getEntryProof(entryNumber, totalEntries)).thenReturn(expectedProof);
+
+        VerifiableLogResource vlResource = new VerifiableLogResource(vlServiceMock);
+        EntryProof actualProof = vlResource.entryProof(entryNumber, totalEntries);
+
+        verify(vlServiceMock, times(1)).getEntryProof(entryNumber, totalEntries);
+        assertThat(actualProof.getProofIdentifier(), equalTo(expectedProof.getProofIdentifier()));
+        assertThat(actualProof.getEntryNumber(), equalTo(expectedProof.getEntryNumber()));
+        assertThat(actualProof.getAuditPath(), containsInAnyOrder(expectedAuditPath.toArray()));
```

Comment:
Should we use is() instead of containsInAnyOrder() because order *does* matter for an entry proof.